File: Лаб_5/Lab_5/Lab_5/TRealNumber.cpp
#include "TRealNumber.h"

TRealNumber::TRealNumber(int integral, int fraction)
{
    this->integral = integral;
    this->fraction = fraction;
}

int TRealNumber::getLastDgt()
{
    return abs(this->fraction)%10;
}

int TRealNumber::getFirstDgt()
{
    int temp_num = abs(this->integral);
    while (temp_num >= 10) {
        temp_num /= 10;
    }
    return temp_num;
}

int TRealNumber::getSumOfDgts()
{
    int sum = 0;
    int temp_num = abs(this->integral);

    while (temp_num) {
        sum += temp_num % 10;
        temp_num /= 10;
    }
    temp_num = this->fraction;
    while (temp_num) {
        sum += temp_num % 10;
        temp_num /= 10;
    }
    return sum;
}

File: Лаб_5/Lab_5_py/TRealNumber.py
from TNumber import *

class TRealNumber(TNumber):
    def __init__(self, integral, fraction):
        self.__integral = integral
        self.__fraction = fraction

    def get_first_digit(self):
        temp_num = abs(self.__integral)
        while temp_num >= 10:
            temp_num //= 10

        return temp_num

    def get_last_digit(self):
        return self.__fraction % 10

    def get_digits_sum(self):
        digits_sum = 0
        # Sum for integer part of number
        temp_num = abs(self.__integral())

        while temp_num:
            digits_sum += temp_num % 10
            temp_num //= 10

        #Sum fo fractional part of number
        temp_num = self.__fraction()

        while temp_num:
            digits_sum += temp_num % 10
            temp_num //= 10

        return digits_sum
